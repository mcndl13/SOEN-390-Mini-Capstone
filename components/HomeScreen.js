// HomeScreen.js
import React from "react";
import {
  View,
  Text,
  StyleSheet,
  TouchableOpacity,
  ScrollView,
} from "react-native";

export default function HomeScreen({ navigation }) {
  return (
    <View style={styles.container}>
      <Text style={styles.title}>Welcome</Text>
      <ScrollView>
        <TouchableOpacity
          style={styles.button}
          onPress={() => navigation.navigate("CampusMap")}
        >
          <Text style={styles.buttonText}>Explore Campus Map</Text>
        </TouchableOpacity>
        <TouchableOpacity
          style={styles.button}
          onPress={() => navigation.navigate("Directions")}
        >
          <Text style={styles.buttonText}>Get Directions</Text>
        </TouchableOpacity>
        <TouchableOpacity
          style={styles.button}
          onPress={() => navigation.navigate("CalendarIntegration")}
        >
          <Text style={styles.buttonText}>Connect to Google Calendar</Text>
        </TouchableOpacity>
        <TouchableOpacity
          style={styles.button}
          onPress={() => navigation.navigate("IndoorDirections")}
        >
          <Text style={styles.buttonText}>Indoor Navigation</Text>
        </TouchableOpacity>
        <TouchableOpacity
          style={styles.button}
          onPress={() => navigation.navigate("PointsOfInterest")}
        >
          <Text style={styles.buttonText}>Find Points of Interest</Text>
        </TouchableOpacity>
      </ScrollView>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
  },
  title: {
    fontSize: 32,
    fontWeight: "semi-bold",
    marginTop: 60,
    marginBottom: 40,
  },
  button: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: "#912338",
    padding: 20,
    margin: 5,
    borderRadius: 30,
  },
  buttonText: {
    color: "white",
    fontSize: 18,
  },
});
