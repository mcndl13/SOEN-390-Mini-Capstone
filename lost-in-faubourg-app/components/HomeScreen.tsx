// HomeScreen.js
import React from "react";
import {
  View,
  Text,
  StyleSheet,
  TouchableOpacity,
  ScrollView,
} from "react-native";

import { NavigationProp } from '@react-navigation/native';

type HomeScreenProps = {
  navigation: NavigationProp<any>;
};

export default function HomeScreen({ navigation }: HomeScreenProps) {
  return (
    <View style={styles.container}>
      <Text style={styles.title}>Welcome</Text>
      <ScrollView>
        <TouchableOpacity
          style={styles.button}
          onPress={() => navigation.navigate("CampusMap")}
        >
          <Text style={styles.buttonText}>Explore Campus Map</Text>
        </TouchableOpacity>
        <TouchableOpacity
          style={styles.button}
          onPress={() => navigation.navigate("Directions")}
        >
          <Text style={styles.buttonText}>Get Directions</Text>
        </TouchableOpacity>
        <TouchableOpacity
          style={styles.button}
          onPress={() => navigation.navigate("CalendarIntegration")}
        >
          <Text style={styles.buttonText}>Connect to Google Calendar</Text>
        </TouchableOpacity>
        <TouchableOpacity
          style={styles.button}
          onPress={() => navigation.navigate("IndoorDirections")}
        >
          <Text style={styles.buttonText}>Indoor Navigation</Text>
        </TouchableOpacity>
        <TouchableOpacity
          style={styles.button}
          onPress={() => navigation.navigate("PointsOfInterest")}
        >
          <Text style={styles.buttonText}>Find Points of Interest</Text>
        </TouchableOpacity>
      </ScrollView>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
  },
  title: {
    fontSize: 32,
    fontWeight: "600",
    marginTop: 60,
    marginBottom: 40,
  },
  button: {
    backgroundColor: "#912338",
    padding: 20,
    margin: 5,
    borderRadius: 30,
    justifyContent: "center",
    alignItems: "center",
  },
  buttonText: {
    color: "white",
    fontSize: 18,
  },
});
