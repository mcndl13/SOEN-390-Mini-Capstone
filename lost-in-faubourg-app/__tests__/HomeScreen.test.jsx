import React from 'react';
import { render, fireEvent, waitFor, act } from '@testing-library/react-native';
import HomeScreen from '../components/HomeScreen';
import { Linking } from 'react-native';
import * as Location from 'expo-location';

const mockNavigation = { navigate: jest.fn() };

jest.mock('expo-location', () => ({
  requestForegroundPermissionsAsync: jest.fn(),
}));

const renderHomeScreen = () => {
  return render(<HomeScreen navigation={mockNavigation} />);
};

describe('HomeScreen', () => {
  beforeAll(() => {
    Location.requestForegroundPermissionsAsync.mockResolvedValue({
      status: 'granted',
    });
  });

  afterAll(() => {
    jest.restoreAllMocks();
  });

  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('renders correctly', async () => {
    const { toJSON } = renderHomeScreen();
    await waitFor(() => {
      expect(toJSON()).toMatchSnapshot();
    });
  });

  const navigationTests = [
    { buttonText: 'Explore Campus Map', target: 'CampusMap' },
    { buttonText: 'Get Directions', target: 'Directions' },
    { buttonText: 'Connect to Google Calendar', target: 'CalendarIntegration' },
    { buttonText: 'Indoor Navigation', target: 'IndoorDirections' },
    { buttonText: 'Find Points of Interest', target: 'PointsOfInterest' },
  ];

  test.each(navigationTests)(
    'navigates to $target when "$buttonText" button is pressed',
    async ({ buttonText, target }) => {
      const { getByText } = renderHomeScreen();
      // Wait for useEffect updates and button to render
      await waitFor(() => getByText(buttonText));
      const button = getByText(buttonText);
      act(() => {
        fireEvent.press(button);
      });
      await waitFor(() => {
        expect(mockNavigation.navigate).toHaveBeenCalledWith(target);
      });
    },
  );

  test('renders title correctly', async () => {
    const { getByText } = renderHomeScreen();
    await waitFor(() => {
      expect(getByText('Welcome')).toBeTruthy();
    });
  });

  test('opens shuttle schedule on button press', async () => {
    const openURLSpy = jest
      .spyOn(Linking, 'openURL')
      .mockImplementation(async () => {});
    const { getByText } = renderHomeScreen();
    await waitFor(() => getByText('Shuttle Schedule'));
    const button = getByText('Shuttle Schedule');
    act(() => {
      fireEvent.press(button);
    });
    await waitFor(() => {
      expect(openURLSpy).toHaveBeenCalledWith(
        'https://www.concordia.ca/maps/shuttle-bus.html#depart',
      );
    });
    openURLSpy.mockRestore();
  });
});