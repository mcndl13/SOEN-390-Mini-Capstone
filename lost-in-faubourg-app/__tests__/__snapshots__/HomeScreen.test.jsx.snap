// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      {
        "fontSize": 32,
        "fontWeight": "600",
        "marginBottom": 40,
        "marginTop": 60,
      }
    }
  >
    Welcome
  </Text>
  <RCTScrollView>
    <View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#912338",
            "borderRadius": 30,
            "justifyContent": "center",
            "margin": 5,
            "opacity": 1,
            "padding": 20,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontSize": 18,
            }
          }
        >
          Explore Campus Map
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#912338",
            "borderRadius": 30,
            "justifyContent": "center",
            "margin": 5,
            "opacity": 1,
            "padding": 20,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontSize": 18,
            }
          }
        >
          Get Directions
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#912338",
            "borderRadius": 30,
            "justifyContent": "center",
            "margin": 5,
            "opacity": 1,
            "padding": 20,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontSize": 18,
            }
          }
        >
          Connect to Google Calendar
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#912338",
            "borderRadius": 30,
            "justifyContent": "center",
            "margin": 5,
            "opacity": 1,
            "padding": 20,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontSize": 18,
            }
          }
        >
          Indoor Navigation
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#912338",
            "borderRadius": 30,
            "justifyContent": "center",
            "margin": 5,
            "opacity": 1,
            "padding": 20,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontSize": 18,
            }
          }
        >
          Find Points of Interest
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
